aspnetcore/blazor/fundamentals/environments.md

```diff
@@ -135,57 +152,208 @@ In the following example for IIS, the custom header (`Blazor-Environment`) is ad
 
-:::moniker-end
+> :::no-loc text="There is no registered service of type 'Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment'.":::
 
-:::moniker range=">= aspnetcore-5.0 < aspnetcore-6.0"
+To address this, create a shared service abstraction and instantiate the service in the server and `.Client` projects with the host environment. The following example demonstrates the approach.
 
-:::code language="razor" source="~/../blazor-samples/5.0/BlazorSample_WebAssembly/Pages/environments/ReadEnvironment.razor" highlight="3,7":::
+> [!NOTE]
+> When using a shared service approach, you can use two distinct service implementations, one for the server project and a separate one for the `.Client` project, or you can use a single service implementation that's configured by server-side or client-side API when the service is created.
+>
+> The demonstration in this section uses a single service implementation:
+>
+> * When the service is instantiated on the server, the host environment is provided by <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder.Environment%2A?displayProperty=nameWithType>.
+> * When the service is instantiated on the client, the host environment is provided by <xref:Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment.Environment%2A?displayProperty=nameWithType>.
+>
+> The service registrations are shown later in this section.
 
-:::moniker-end
+Add a shared class library project for the Blazor Web App:
 
-:::moniker range="< aspnetcore-5.0"
+* Visual Studio: Right-click the Blazor Web App's solution file in **Solution Explorer**. Select **Add** > **New Project** > **Class Library**. Use the name of the server project with "`.Shared`" added to the end of the name (for example: `BlazorWebAppSample.Shared`).
+* Visual Studio Code/.NET CLI: Execute `dotnet new classlib -o {PROJECT NAME}` from a command prompt in the solution's folder. The `-o|--output` option creates a folder for the class library and names the project. For the `{PROJECT NAME}` placeholder, use the name of the server project with "`.Shared`" added to the end of the name (for example: `BlazorWebAppSample.Shared`).
+
+Create a project reference for the class library in both the server and `.Client` projects:
 
-:::code language="razor" source="~/../blazor-samples/3.1/BlazorSample_WebAssembly/Pages/environments/ReadEnvironment.razor" highlight="3,7":::
+* Visual Studio: Right-click each project and select **Add** > **Project Reference**. Select the checkbox for the shared project and select **OK**.
+* Visual Studio Code/.NET CLI: Execute `dotnet add reference {PATH}` in a command shell from each of the server and `.Client` project folders. The `{PATH}` placeholder is the path to the shared class library project.
+
+In the shared (`.Shared`) project:
+
+* Add an interface for the service abstraction.
+* Add classes for the service implementation and host environment extensions.
```

Comment:
Yes, I was split in my mind on if that felt good, and that is what you said in the email.

I just narrowly was leaning in the direction of portability, using a class lib or package across BWAs. Also, it feels like we're dealing with two distinct projects that are sharing a central resource, even if that isn't the best way to think about it.

However, it would be simpler to explain that way ... in the `.Client` project, so let's shorten this up with that approach. I'll make the changes.